lib/webpacker/compiler.rb

```diff
@@ -13,13 +13,17 @@ def initialize(webpacker)
 
   def compile
     if stale?
+      clobber
       record_compilation_timestamp
       run_webpack
-    else
-      logger.debug "No compiling needed as everything is fresh"
     end
   end
 
+  def clobber
+    config.public_output_path.rmtree if config.public_output_path.exist?
+    config.cache_path.rmtree if config.cache_path.exist?
+  end
```

Comment:
@dhh Not needed if we are using dev server but if we are using on-demand compilation the output folder can get pretty big after each compilation since we are now using assets hashed. Although, this doesn't solve the same problem when using watcher.